docs/examples/101/application-gateway-v2-autoscale-create/main.bicep

```diff
@@ -0,0 +1,156 @@
+param virtualNetworkName string = 'Application-Vnet'
+param vnetAddressPrefix string = '10.0.0.0/16'
+param subnetName string = 'ApplicationGatewaySubnet'
+param subnetPrefix string = '10.0.0.0/24'
+param applicationGatewayName string = 'applicationGatewayV2'
+param minCapacity int = 2
+param maxCapacity int = 10
+param frontendPort int = 80
+param backendPort int = 80
+param backendIPAddresses array {
+  default: [
+    {
+      IpAddress: '10.0.0.4'
+    }
+    {
+      IpAddress: '10.0.0.5'
+    }
+  ]
+}
+param cookieBasedAffinity string {
+  default: 'Disabled'
+  allowed: [
+    'Enabled'
+    'Disabled'
+  ]
+}
+param location string = resourceGroup().location
+
+var appGwPublicIpName = '${applicationGatewayName}-pip'
+
+resource virtualNetwork 'Microsoft.Network/virtualNetworks@2020-06-01' = {
+  name: virtualNetworkName
+  location: location
+  properties: {
+    addressSpace: {
+      addressPrefixes: [
+        vnetAddressPrefix
+      ]
+    }
+    subnets: [
+      {
+        name: subnetName
+        properties: {
+          addressPrefix: subnetPrefix
+        }
+      }
+    ]
+  }
+}
+
+resource publicIP 'Microsoft.Network/publicIPAddresses@2020-06-01' = {
+  name: appGwPublicIpName
+  location: location
+  sku: {
+    name: 'Standard'
+  }
+  properties: {
+    publicIPAllocationMethod: 'Static'
+  }
+}
+
+resource applicationGateway 'Microsoft.Network/applicationGateways@2020-06-01' = {
+  name: applicationGatewayName
+  location: location
+  properties: {
+    sku: {
+      name: 'Standard_v2'
+      tier: 'Standard_v2'
+    }
+    autoscaleConfiguration: {
+      minCapacity: minCapacity
+      maxCapacity: maxCapacity
+    }
+    gatewayIPConfigurations: [
+      {
+        name: 'appGatewayIpConfig'
+        properties: {
+          subnet: {
+            id: resourceId('Microsoft.Network/virtualNetworks/subnets', virtualNetworkName, subnetName)
```

Comment:
use `virtualNetwork.name` and you should be able to get rid of the `dependsOn` on line 153